src/Layouts/TeacherLayout.tsx

```diff
@@ -0,0 +1,227 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Outlet, useNavigate } from "react-router";
+import home_nav_icon from "../assets/Images/Home_nav_icon.png";
+import teachers_nav_icon from "../assets/Images/teachers_nav_icon.png";
+import teachers_nav_icon_active from "../assets/Images/teachers_nav_icon_active.png";
+import classes_nav_icon from "../assets/Images/Classes_nav_icon.png";
+import classes_nav_icon_active from "../assets/Images/Classess_nav_icon_active.png";
+import profile_nav_icon from "../assets/Images/Profile_nav_icon.png";
+import profile_nav_icon_active from "../assets/Images/profile_nav_icon_active.png";
+
+import contact_nav_icon from "../assets/Images/Contact_nav_icon.png";
+import logout_nav_icon from "../assets/Images/Logout_nav_icon.png";
+import bell_icon from "../assets/Images/Bell_icon.png";
+
+import NavItems from "../Components/Elements/NavItems";
+import { useContext, useEffect, useState } from "react";
+import Counter from "../Components/Elements/Counter";
+import SearchBox from "../Components/Elements/SearchBox";
+import Profile from "../Components/Elements/Profile";
+import menu_icon from "../assets/Images/men_icon.png";
+import close_icon from "../assets/Images/close_icon.png";
+import UserContext from "../contexts/UserContext";
+import { baseURL } from "../const/const";
+import axios from "axios";
+const TeacherLayout = () => {
+  const [notifications, setNotifications] = useState(5);
+  const [search, setSearch] = useState("");
+  const [active, setActive] = useState("active");
+  const [mobileMenuActive, setMobileMenuActive] = useState(false);
+  const { user, setUser } = useContext(UserContext);
+  const navigate = useNavigate();
+  const logout = async () => {
+    const result = await axios.get(baseURL + "/api/v1/auth/logout", {
+      withCredentials: true,
+    });
+    if (result.status === 200) {
+      setUser(result.data);
+      navigate("/teacher/login");
+    }
+  };
+  useEffect(() => {
+    if (!user.first_name) {
+      navigate("/teacher/login");
+    }
+  });
```

Comment:
UseEfect logic needs to be updated with a dependency Array, if you need this run only once, add a custom boolean property to dep Array that reruns the useEffect Logic

Reply:
noted. I'll change it

Reply:
Fixed